pkg/reconciler/pipelinerun/resources/pipelineref_test.go

```diff
@@ -628,41 +637,65 @@ func TestGetVerifiedPipelineFunc_VerifyError(t *testing.T) {
 		t.Fatal("fail to sign pipeline", err)
 	}
 
-	tamperedPipeline := signedPipeline.DeepCopy()
-	tamperedPipeline.Annotations["random"] = "attack"
-	tamperedPipelineBytes, err := json.Marshal(tamperedPipeline)
+	modifieddPipeline := signedPipeline.DeepCopy()
+	modifieddPipeline.Annotations["random"] = "attack"
+	modifiedPipelineBytes, err := json.Marshal(modifieddPipeline)
 	if err != nil {
 		t.Fatal("fail to marshal pipeline", err)
 	}
-	resolvedTampered := test.NewResolvedResource(tamperedPipelineBytes, nil, sampleConfigSource.DeepCopy(), nil)
-	requesterTampered := test.NewRequester(resolvedTampered, nil)
+	resolvedModified := test.NewResolvedResource(modifiedPipelineBytes, nil, sampleConfigSource.DeepCopy(), nil)
+	requesterModified := test.NewRequester(resolvedModified, nil)
 
 	pipelineRef := &v1beta1.PipelineRef{ResolverRef: v1beta1.ResolverRef{Resolver: "git"}}
 
 	testcases := []struct {
-		name                     string
-		requester                *test.Requester
-		resourceVerificationMode string
-		expected                 runtime.Object
-		expectedErr              error
+		name                      string
+		requester                 *test.Requester
+		verificationNoMatchPolicy string
+		expected                  runtime.Object
+		expectedErr               error
 	}{
 		{
-			name:                     "unsigned pipeline with enforce policy",
-			requester:                requesterUnsigned,
-			resourceVerificationMode: config.EnforceResourceVerificationMode,
-			expected:                 nil,
-			expectedErr:              trustedresources.ErrResourceVerificationFailed,
+			name:                      "unsigned pipeline fails verification with fail no match policy",
```

Comment:
It looks like this tests a case where the policy is set to "fail" and there are matching policies, but the pipeline fails verification. Is there a test case for "fail" with no matching policies?

Reply:
yes, it is a missing case, added!

Reply:
which test case is this? I'm not seeing it

Reply:
Oh I thought I added this one but looks like I only added these cases and was distracted by an urgent issue last week
https://github.com/tektoncd/pipeline/compare/549a302e3fcced0360848bbf5a1078ff3aff21fb..c31dde1f124484d140d117ec3ff04749bd0cff50

Just added in the latest commit. Thanks for the reminder!

Reply:
Thanks for following up! I can see you've also made the changes in taskref_test.go, thanks!